datafusion/core/src/physical_plan/sorts/merge.rs

```diff
@@ -51,7 +54,9 @@ macro_rules! merge_helper {
     }};
 }
 
-/// Perform a streaming merge of [`SendableRecordBatchStream`]
+/// Perform a streaming merge of [`SendableRecordBatchStream`] based on provided sort expressions
+/// while preserving order. This is a convenience wrapper for [`SortPreservingMergeStream`] and
+/// chooses a right cursor for the expressions and the data type
```

Comment:
Not sure about the cursor. You mean `RowCursorStream`?

Reply:
I think we can skip cursor in the doc and only mention this merges `SendableRecordBatchStream`s by sorting and preserving the order.

Reply:
This function switches between `RowCursorStream` and `FieldCursorStream` depending on the data type.
`SortPreservingMergeStream` expects a cursor and that's what this wrapper provides as convenience

Reply:
Removed the cursor explanation.

Reply:
I feel it is too much detailed to the implementation. No strong option here. It's fine.